syntax = "proto3";

message Account {
  string account_number = 1;
  string email_id = 2;
  string account_type = 3;
  string address = 4;
  string ssn_number = 5;
  string government_id = 6;
  string first_name = 7;
  string last_name = 8;
  string currency = 9;
  double balance = 10;
}


message CreateAccountRequest {
  string email_id = 1;
  string account_type = 2;
  string address = 3;
  string ssn_number = 4;
  string government_id = 5;
  string first_name = 6;
  string last_name = 7;
}

message CreateAccountResponse {
  bool result = 1;
}

message GetAccountsRequest {
  string email_id = 1;
}

message GetAccountsResponse {
  repeated Account accounts = 1;
}




message AccountDetail{
  string account_number = 1;
  string account_holder_name = 2;
  double balance = 3;
  string currency = 4;
}

message GetAccountDetailRequest {
  string account_number = 1;
}

message GetAccountDetailResponse {
  AccountDetail account = 1;
}





service AccountDetailsService {
  rpc getAccountDetails(GetAccountDetailRequest) returns (GetAccountDetailResponse);
  rpc createAccount(CreateAccountRequest) returns (CreateAccountResponse);
  rpc getAccounts(GetAccountsRequest) returns (GetAccountsResponse);
}



